### PR TITLE
Implement Latest Gw2Sharp (0.9.3)

### PR DESCRIPTION
**Pull request by dlamkins**
This PR works to reimplement Gw2Sharp to utilize it for the Mumble service and to update our WebApi service utilization.

Breaking:
- Camera and Player service have been removed and fields have been restructured under the Mumble service itself.
- Fields on the WebApi service have been updated.
- UiSize is no longer exposed on the GraphicsService and is instead listed under the Mumble services' UI field.
- Gw2ApiManager exposes the root WebApi client instead of V2 directly now.
- ChatCode util was removed under the `Common.Gw2` namespace since Gw2Sharp supports generating and parsing these directly.

### CODE REVIEW

**Comment by greaka** [comment]
since this is some pretty big milestone, and also includes some breaking changes, id like to propose bumping the version to 0.5 with this PR

**Comment by dlamkins** [comment]
> since this is some pretty big milestone, and also includes some breaking changes, id like to propose bumping the version to 0.5 with this PR

Agreed!  🙂 

Also noting here that we'll need to retarget again to the new 9.3 release.  Doesn't appear to have anything breaking coming from the 9.1 or 9.2 we're currently targeting.

https://github.com/Archomeda/Gw2Sharp/releases/tag/v0.9.3

File: Blish HUD/GameServices/ContentService.cs
Comment (greaka) [inline Blish HUD/GameServices/ContentService.cs:115]: Why don't we use the ArchiveCacheMethod here?

File: Blish HUD/GameServices/Gw2Mumble/PlayerCamera.cs
Comment (greaka) [inline Blish HUD/GameServices/Gw2Mumble/PlayerCamera.cs]: Do you want modules to be able to set this? Maybe we want an internal set and have it exposed somewhere if needed.

File: Blish HUD/GameServices/Gw2Mumble/PlayerCamera.cs
Comment (greaka) [inline Blish HUD/GameServices/Gw2Mumble/PlayerCamera.cs]: Put the clamp into the Update function, so it doesn't need to be calculated whenever someone reads it. We need it at least once anyway, so we can only gain here.

File: Blish HUD/GameServices/Gw2WebApi/ManagedConnection.cs
Comment (greaka) [inline Blish HUD/GameServices/Gw2WebApi/ManagedConnection.cs]: I think there is an excess whitespace in there, or is it intentional? Why?

File: Blish HUD/GameServices/Modules/Managers/Gw2ApiManager.cs
Comment (greaka) [inline Blish HUD/GameServices/Modules/Managers/Gw2ApiManager.cs:14]: I would like to propose that we use another naming convention for static fields, so it is clear in the code.

File: Blish HUD/GameServices/Modules/Managers/Gw2ApiManager.cs
Comment (greaka) [inline Blish HUD/GameServices/Modules/Managers/Gw2ApiManager.cs]: You shouldn't need to call `.ToArray()` here.

**Review by greaka — CHANGES_REQUESTED** [review]

**Comment by greaka** [comment]
This PR is also missing the version change to 0.5, I think it should be changed before we merge it.

File: Blish HUD/GameServices/Gw2WebApi/ManagedConnection.cs
Comment (dlamkins) [reply Blish HUD/GameServices/Gw2WebApi/ManagedConnection.cs]: This was due to this:
https://discordapp.com/channels/384735285197537290/589031897963692034/692548369742168084

> So taking a closer look at the UA appending that is done, a space is not placed in between.  This causes the UA format to be incorrect (I believe due to having multiple / without being split by a  ).  Appending my provided UA with a space allows it to work.

It appears that this was fixed in 9.2, though, so I'll get that extra space removed. :)

https://github.com/Archomeda/Gw2Sharp/commit/022d7296cc3dc64c883116105987a626136edc74

File: Blish HUD/GameServices/Modules/Managers/Gw2ApiManager.cs
Comment (dlamkins) [reply Blish HUD/GameServices/Modules/Managers/Gw2ApiManager.cs:14]: Currently we're using _camelCase for all non-public fields.  Naming convention here is indicating public vs. not public, which is fairly common from what I can tell.

We can certainly discuss potential alternatives, but I don't think this PR should be the place for that discussion or ground zero for those changes.

File: Blish HUD/GameServices/Modules/Managers/Gw2ApiManager.cs
Comment (dlamkins) [reply Blish HUD/GameServices/Modules/Managers/Gw2ApiManager.cs]: Good catch.

**Comment by dlamkins** [comment]
FYI - CI has the final say on the version which is why the assembly version itself hasn't been updated from 4.0 in quite a while.  It'll inject and override whatever is specified in the project file with the incrementing build value included.

That being said, I've gone ahead and upped the project's assembly version as requested.

File: Blish HUD/GameServices/ContentService.cs
Comment (dlamkins) [reply Blish HUD/GameServices/ContentService.cs:115]: I've had a mixed experience with `ArchiveCacheMethod` in the past and troubles getting it to flush to the disk correctly causing it to not keep a cache of things requested that session until the session ends and it flushes to disk.  This causes all requests to get re-requested every request.  There were recent changes to it after some other folks brought up having similar issues, so I'll need to review those.

Ultimately, I need more time to review the current implementation in Gw2Sharp to ensure it works correctly when we use it and will have a PR to update this to an `ArchiveCacheMethod` once I have done that.  Using the `MemoryCacheMethod` for now will at least cover caching for the active session until that work is done.

File: Blish HUD/GameServices/Gw2Mumble/PlayerCamera.cs
Comment (dlamkins) [reply Blish HUD/GameServices/Gw2Mumble/PlayerCamera.cs]: I personally have no preference either way.  It was a holdover from the original `CameraService` and was kind of left like that in case any modules wanted to play with them.  

I can't think of any strong use cases for doing that, though, so for now I've removed the public setters on them and if we ever want to make them accessible down the road we can add them back or provide some other method of exposing them without causing a breaking change.

File: Blish HUD/GameServices/Modules/Managers/Gw2ApiManager.cs
Comment (greaka) [reply Blish HUD/GameServices/Modules/Managers/Gw2ApiManager.cs:14]: agreed

**Review by greaka — DISMISSED** [review]

**Comment by dlamkins** [comment]
Sorry for the re-review.  I removed the, now redundant, ChatCode utility as Gw2Sharp can now take care of that.

**Review by greaka — APPROVED** [review]

File: Blish HUD/GameServices/Gw2Mumble/PlayerCamera.cs
Comment (dlamkins) [inline Blish HUD/GameServices/Gw2Mumble/PlayerCamera.cs:58]: Fixes #145!

**Pull request merged**
